### ISSUE TITLE
client.write prevents other code from executing

### ISSUE THREAD

**Issue by facekapow** [kesla/runtime-node-net#11]
When using `client.write` in the following example, the rest of the code in the callback is not executed:

``` js
const net = require('net');
  var server = net.createServer(function(socket) {
    var chunks = '';
    socket.on('data', function(chunk) {
      chunks += chunk;
      runtime.tty.print(chunks);
    });
    socket.on('end', function() {
      // this never gets printed
      runtime.tty.print('ending');
      server.close();
    });
  });
  // Port 23 is the accepted standard for telnet
  server.listen(23, function() {
    var client = net.createConnection({ port: 23 }, function() {
      // outputs "stuff"
      client.write('stuff\r\n');
      // this isn't run
      client.write('hey\r\n');
      // neither is this
      client.end();
    });
  });
```

The code is supposed to print:

```
stuff
hey
ending
```

But I get this:

```
stuff
```

Does `client.write` block? Is it synchronous? If not, any idea what it could be?

**Comment by facekapow** [comment]
@kesla I ran a test with plain `runtime` `TCPSocket`s and I was able to write multiple times with the same socket. BUT, doing the same with `runtime-node-net` I was NOT able to write multiple times.

**Comment by iefserge** [comment]
hm, it shouldn't block because there are no blocking functions in runtime and nothing to block on :)

**Comment by facekapow** [comment]
@iefserge Well, it could be something with `node`'s `stream` duplex, which is used in `socket.js`, because, like I said, raw `runtime` `TCPSocket`s and `TCPServerSocket`s allow me to write multiple times on the same socket.

**Comment by kesla** [comment]
@ArielAbreu hey, sorry for slow response. Taking a look at this now!

**Comment by kesla** [comment]
@ArielAbreu Thank you so much! This has been fixed in v0.1.4 of this library.

There's still a bunch of missing functionality missing in this wrapper - and I'm going to implement the missing pieces step by step - so having bug reports like this is great!

**Comment by facekapow** [comment]
No problem! Now I can rewrite my telnet server for runtime to use runtime-node-net! Thanks a bunch for this fix!
